src/App.tsx

```diff
@@ -1,19 +1,152 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+import { Todo } from './components/Todo';
+import { ToDoType } from './api/type';
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+interface State {
+  copyTodos: ToDoType[],
+  title: string,
+  userId: number,
+  noUser: boolean,
+  noTitle: boolean,
+}
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
-};
+export class App extends React.Component <{}, State> {
+  state = {
+    copyTodos: [...todos],
+    title: '',
+    userId: 0,
+    noUser: false,
+    noTitle: false,
+  };
 
-export default App;
+  handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      userId: Number(value),
+      noUser: false,
+    });
+  };
+
+  handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      title: value,
+      noTitle: false,
+    });
+  };
+
+  submitForm = () => {
+    const {
+      title, userId, noUser, noTitle,
+    } = this.state;
+
+    if (userId !== 0 && title !== '') {
+      this.setState((state) => ({
+        copyTodos:
+        [
+          ...state.copyTodos,
+          {
+            userId: +userId,
+            title,
+            id: state.copyTodos.length + 1,
+            completed: false,
+          },
+        ],
+      }));
+
+      this.setState({
+        title: '',
+        userId: 0,
+      });
```

Comment:
You can do it in one setState
```suggestion
      this.setState((state) => ({
        title: '',
        userId: 0,
        copyTodos: [
          ...state.copyTodos,
          {
            userId: +userId,
            title,
            id: state.copyTodos.length + 1,
            completed: false,
          },
        ],
      }));
```